### ISSUE TITLE
change order of images

### ISSUE THREAD

**Issue by wolle404** [spatie/laravel-medialibrary#382]
Hey there,

I was wondering why there's a `order_column` column.
Can this column be used for a custom order?
Or should I have to implement it all by my self using the properties?

Thanks!

**Comment by nicolasbeauvais** [comment]
I couldn't find any documentation on the `order_column`

Most of the code that handles order is in the `Spatie\MediaLibrary\SortableTrait` which is by default included in the `Media` class. Also, the base `MediaObserver` is used to set the `order_column` when creating a media.

By looking at this I can understand that: by default every newly created media as an `order_column` that correspond to last inserted media order + 1 (across all media, independent of the file collection / disk).

Also, there is a `Media::setNewOrder` static function:

```php
 /**
  * This function reorders the records: the record with the first id in the array
  * will get order 1, the record with the second it will get order 2, ...
  *
  * A starting order number can be optionally supplied (defaults to 1).
  *
  * @param array $ids
  * @param int   $startOrder
  */
```

By default, medias are ordered using the `order_column`.

@freekmurze do you have a little more details on this ? I could write some doc for it using what is learned in this issue.

**Comment by freekmurze** [comment]
The functionality around  `order_column` should be indeed be documented. @nicolasbeauvais correctly described what it does. In short this allows you to get all media in the same order as it was added to the medialibrary. You can set a new order manually by using `Media::setNewOrder` or by manually setting the value of `order_column`.

@nicolasbeauvais I'd certainly accept some documentation around this small feature.

**Comment by amosmos** [comment]
@freekmurze - Would agree that a better\sensible use case for order_column is to order the files per each model and each collection separately? If so would it be something you would consider adding to the package or at least accepting a PR that does it?

**Comment by freekmurze** [comment]
I don't think generating separate values per model and collection are needed. Just using the maximum value or the order column + 1 seems fine to me. I don't mind gaps in the order column when retrieve media.

Could you describe an specific use case where you think the values of the order column are causing problems?

**Comment by amosmos** [comment]
There is no doubt that the default order is ok and working. But the problem happens when we want to change the order - I think this requires either sending the entire collection of medias to updateMedia in order to re-order them, or to have your own ordering mechanism and set the order_column value "manually".



I think that if you keep the order based on each model and each collection you gain few benefits:
1. It is better to understand and better represent the most common use case for this field.
2. It makes re-ordering easier. For just a very simple example: I can show my users an "order" field, filled with the order_column value, under each media, and let them manually choose the order value. Of course I can also do it now and it will work, but it will seem very strange for the users to see it (they will see "random" numbers for ordering).

To sum up - it just looks like you added this column for a reason, but kinda "stopped in the middle" once you got some sort of a "working level solution". It's just doesn't seem complete... While it seems like it will probably not be to complicated to complete (and of course I'm willing to try myself if you prefer).

I hope so much you understand this is by no means criticism, I am SO grateful for your work on this package and all the others and I use them A LOT (expect a postcard). Just had to share my feeling about this little aspect and offer my assistance should you agree it is something that can be approved.

**Comment by plakhin** [comment]
Hello!
Faced the same issue, thought order_column is an order for each model, i.e. I can have several images for each post and rearrange their order...

> We pass those id's to the `updateMedia` function.

Didn't find anything regarding this function in docs. Would you be so kind to describe how do you use it a bit more detailed?

Thank you.

**Comment by amosmos** [comment]
@plakhin I think the way it works is that you send it a list of media id's and the function re-order them in the order that you sent them.

**Comment by GoodJob** [comment]
The photo uploader field automatically displays dynamically uploaded photos with no feature to order them. How to turn off that?

For users to order photos, an additional photo gallery view should be added, custom made, which will double the existing one gallery view by spatie which appears after photos uploaded

So how to solve this? 

Thank you for nice product 🙏
